src/Temporalio/Client/TestService.cs

```diff
@@ -0,0 +1,41 @@
+using System.Threading.Tasks;
+using Google.Protobuf;
+
+namespace Temporalio.Client
+{
+    /// <summary>
+    /// Raw gRPC test service.
+    /// </summary>
+    public abstract partial class TestService : RpcService
+    {
+        internal override Bridge.Interop.RpcService Service => Bridge.Interop.RpcService.Test;
+
+        /// <summary>
+        /// Implementation of the test service.
+        /// </summary>
+        public class Impl : TestService
+        {
+            private readonly TemporalConnection connection;
+
+            /// <summary>
+            /// Create an implementation of the test service.
+            /// </summary>
+            /// <param name="connection">Connection to use.</param>
+            public Impl(TemporalConnection connection)
+            {
+                this.connection = connection;
+            }
+
+            /// <inheritdoc />
+            protected async override Task<T> InvokeRpcAsync<T>(
```

Comment:
I built it this way so if you wanted to provide your own `WorkflowService` implementation with your own `InvokeRpcAsync` call that maybe wrapped the real one, you could without a connection. This is also why this call `protected` and not `internal` and why the service class is abstract and this is in a nested `Impl` class. Otherwise I wouldn't have any of this exposed, but this is reasonable from an OO POV.